vllm/engine/metrics.py

```diff
@@ -29,41 +27,47 @@
 
 # begin-metrics-definitions
 class Metrics:
+    """
+    vLLM uses a multiprocessing-based frontend for the OpenAI server.
+    This means that we need to run prometheus_client in multiprocessing mode
+    See https://prometheus.github.io/client_python/multiprocess/ for more
+    details on limitations.
+    """
     labelname_finish_reason = "finished_reason"
     _gauge_cls = prometheus_client.Gauge
     _counter_cls = prometheus_client.Counter
     _histogram_cls = prometheus_client.Histogram
 
     def __init__(self, labelnames: List[str], max_model_len: int):
-        # Unregister any existing vLLM collectors
-        self._unregister_vllm_metrics()
-
-        # Config Information
-        self._create_info_cache_config()
```

Comment:
Cannot support `Info` metric with multiprocessing

Reply:
info type metrics are just syntactic sugar for a gauge permanently set to 1

```
LIBRARY_VERSION_INFO = pc.Gauge(
    namespace="somelibrary",
    name="info",
    labelnames=("version", "major", "minor", "patchlevel"),
    documentation="our library version information",
    multiprocess_mode="mostrecent",
)
```

Just replace them with gauge metrics and set them to 1 on creation.

Reply:
I think it's kind of silly that the prometheus client library decided to just drop support entirely. When the `multiprocess_mode` flag just controls how the value of the metric is aggregated and the value of info metrics is irrelevant.

Reply:
Agree!